app/src/components/ConversationHistory.tsx

```diff
@@ -4,55 +4,58 @@ import { Trash, ArrowRight } from "lucide-react";
 import { AnimatePresence, motion } from "framer-motion";
 import { Button } from "./ui/button";
 
-export interface HistoryChat {
+export interface Conversation {
     id: number;
     created_at: string;
 };
 
-export default function ChatsHistory({
+export default function ConversationHistory({
     supabaseClient,
     handleClose,
     fetchLastConversation,
 }: {
     supabaseClient: SupabaseClient;
     handleClose: () => void;
-    fetchLastConversation: (chatId: number) => void;
+    fetchLastConversation: (conversationId: number) => void;
 }) {
-    const [listReceived, setListReceived] = useState(false);
-    const [chatsList, setChats] = useState<HistoryChat[]>([]);
+    const [conversations, setConversations] = useState<Conversation[]>([]);
 
-    const fetchChatsList = async () => {
+    const removeConversation = async (conversationId: number) => {
         try {
-            const { data, error } = await supabaseClient
+            const { error } = await supabaseClient
                 .from("conversations")
-                .select("id, created_at")
-                .order("created_at", { ascending: false });
-            if (data) {
-                setChats(data);
-            }
+                .delete()
+                .eq("id", conversationId);
         } catch (error: any) {
             console.error("ERROR", error);
-        }
-    };
-
-    const removeChat = async (chatId: number) => {
-        try {
+        } finally {
             const { data, error } = await supabaseClient
                 .from("conversations")
-                .delete()
-                .eq("id", chatId);
+                .select("id, created_at")
+                .order("created_at", { ascending: false });
             if (data) {
-                setChats(data);
+                setConversations(data);
             }
-        } catch (error: any) {
-            console.error("ERROR", error);
         }
-        await fetchChatsList();
     }
 
     useEffect(() => {
-        fetchChatsList();
-    }, []);
+        const fetchConversationList = async () => {
+            try {
+                const { data, error } = await supabaseClient
+                    .from("conversations")
+                    .select("id, created_at")
+                    .order("created_at", { ascending: false });
+                if (data) {
+                    setConversations(data);
+                }
+            } catch (error: any) {
+                console.error("ERROR", error);
+            }
+        };
+        
+        fetchConversationList();
+    }, [supabaseClient]);
```

Comment:
I'd suggest we use [react-query](https://tanstack.com/query/latest) or [swr](https://swr.vercel.app/) for requests like these.

Reply:
Agreed, I was trying to clean those up w/o going too deep since the focus was the nav menu popover

Reply:
I am all for using react query, but if we use SWR, then we won't be able to run it as a native mobile app - fyi

Reply:
@Krupskis do you have experience with that, would you wanna open a PR?

Reply:
I am fine with react query. @Jacksonmills busy with RAG stuff now, do you wanna try it and I can take a look? I can send some code examples to you

Reply:
@adamcohenhillel why doesn't SWR work on native apps?

Reply:
I will play around with it tonight/tomorrow, maybe in a separate PR

Reply:
@Krupskis I looked at it in the past, but not entirely sure. I think this is because with SWR you have the server rendering your components, meaning you need to have the nextjs server running somewhere, which is not so compatiable with the way Capacitor works (compiling to native, and server as a regular app) 

Basically - you need to have all the components when you compile it to native, something you don't have with SWR

might be wrong here though

Reply:
@Krupskis https://github.com/adamcohenhillel/ADeus/pull/37